internal/module/services/version_manager/version_manager.go

```diff
@@ -104,6 +119,30 @@ func (s *VersionManagerService) BeginVersion(ctx context.Context, request *terra
 		return nil, CreateModuleVersionError
 	}
 
+	parsedVersion := versions.MustParseVersion(strings.ReplaceAll(request.Module.GetVersion(), "v", ""))
```

Comment:
Do not use `MustParseVersion` here - if the version doesn't parse it will cause a `panic()`.
Instead use `ParseVersion()`
```suggestion
	if parsedVersion, err := versions.ParseVersion(strings.ReplaceAll(request.Module.GetVersion(), "v", "")); err != nil {
```